crates/kubelet/src/fs_watch/mod.rs

```diff
@@ -0,0 +1,306 @@
+//! A simple abstraction layer over OS specific details on watching a filesystem. Due to a bug in
+//! MacOS with sending an event on socket creation, we need to implement our own hacky watcher. To
+//! keep it as clean as possible, this module abstracts those details away behind a `Stream`
+//! implementation. A bug has been filed with Apple and we can remove this if/when the bug is fixed
```

Comment:
Would you mind adding a tracking ID to determine when we can remove this file?

Reply:
It is in the PR description, but it is also non-public (couldn't find a setting for it). Do you still want it here?

Reply:
Yes please! That way it makes it easier to track down the line. I've found that historical references being stored in code makes it easier to track several years down the line in case it's never fixed upstream. Just something like "Tracking ID FB8830541 - @thomastaylor312 can look up the status of the bug".